lisa/tools/lsvmbus.py

```diff
@@ -115,25 +119,52 @@ def _check_exists(self) -> bool:
             _exists = super()._check_exists()
             if not _exists:
                 self._command = previous_command
+
+        if _exists and isinstance(self._linux_os, Ubuntu):
+            # fix for issue happen on Canonical UbuntuServer 16.04-LTS 16.04.201703020
+            # lsvmbus exists by default, but it doesn't give expected results, still
+            # download lsvmbus src
+            # WARNING: lsvmbus not found for kernel 4.4.0-65
+            #   You may need to install the following packages for this specific kernel:
+            #     linux-tools-4.4.0-65-generic
+            #     linux-cloud-tools-4.4.0-65-generic
+            #   You may also want to install one of the following packages to keep up to
+            # date:
+            #     linux-tools-generic
+            #     linux-cloud-tools-generic
+            cmd_result = self.run("-vv", shell=True)
+            if self.__pattern_not_found.match(cmd_result.stdout):
+                _exists = False
+
         return _exists
 
+    def _install_from_src(self) -> None:
+        wget_tool = self.node.tools[Wget]
+        file_path = wget_tool.get(self._lsvmbus_repo, "$HOME/.local/bin")
+        # make the download file executable
+        self.node.execute(f"chmod +x {file_path}")
+        self._command = file_path
+
     def install(self) -> bool:
-        linux_os: Linux = cast(Linux, self.node.os)
         package_name = ""
-        if isinstance(linux_os, Redhat):
+        if isinstance(self._linux_os, Redhat):
             package_name = "hyperv-tools"
-        elif isinstance(linux_os, Suse):
+            self._linux_os.install_packages(package_name)
+        elif isinstance(self._linux_os, Suse):
             package_name = "hyper-v"
-        elif isinstance(linux_os, Ubuntu) and not isinstance(linux_os, Debian):
+            self._linux_os.install_packages(package_name)
+        elif isinstance(self._linux_os, Ubuntu):
             package_name = "linux-cloud-tools-common"
-        if package_name:
-            linux_os.install_packages(package_name)
-        else:
-            wget_tool = self.node.tools[Wget]
-            file_path = wget_tool.get(self._lsvmbus_repo, "$HOME/.local/bin")
-            # make the download file executable
-            self.node.execute(f"chmod +x {file_path}")
-            self._command = file_path
+            self._linux_os.install_packages(package_name)
+            cmd_result = self.run("-vv", shell=True)
+            if self.__pattern_not_found.match(cmd_result.stdout):
```

Comment:
Add one-line comment, let people refer to comments in `_check_exists`.

Reply:
Sorry, I don't understand.

Reply:
Since those lines are similar with what in `_check_exists`, but when people look at here only, they may not know why yet. To avoid duplicating the whole long comments, you can just write one line like `this logic is to handle Ubuntu xxx behavior, refer to similar logic in _check_exists above.`